modules/core/test/test_mat.cpp

```diff
@@ -1248,6 +1248,58 @@ static void check_ndim_shape(const cv::Mat &mat, int cn, int ndims, const int *s
         EXPECT_EQ(mat.size[i], sizes[i]);
 }
 
+TEST(Core_Mat, convertTo_inplace_8UC3_8SC3)
+{
+    cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
+    void* p = A.data;
+    int cn = A.channels();
+    int depth = CV_8S;
+    int dtype = CV_MAKE_TYPE(depth, cn);
+    A.convertTo(A, depth);
+    EXPECT_EQ(p, A.data);
+    EXPECT_EQ(A.type(), dtype);
```

Comment:
Add a check for resulting values:
```cpp
EXPECT_EQ(0, countNonZero(A != cv::Scalar(1, 2, 3)));
```

Reply:
I had add a function to replace countNonZero, because countNonZero only support cn == 1.